### PR TITLE
setup test framework

### PR DESCRIPTION
**Pull request by elharo**
@briandealwis @akerekes I'm trying to get a simple test to run. However I'm missing something because I get the error message

[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: com.google.gcp.appengine.localserver.test 94.0.0.qualifier
[ERROR]   Missing requirement: com.google.gcp.appengine.localserver.test 94.0.0.qualifier requires 'bundle com.google.gcp.appengine.localserver 0.0.0' but it could not be found
[ERROR] 

I suspect this is a one line fix. Anyone see what I'm missing?

### CODE REVIEW

**Comment by elharo** [comment]
Looks like this is the problem

Fragment-Host: com.google.gcp.appengine.localserver;bundle-version="94.0.0"

I can't figure out how to change that to pick up the -SNAPSHOT qualifier.

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Manifest-Version: 1.0
+Bundle-ClassPath: .
+Bundle-ManifestVersion: 2
+Bundle-Name: Test
+Bundle-RequiredExecutionEnvironment: JavaSE-1.7
+Bundle-SymbolicName: com.google.gcp.appengine.localserver.test
```
Comment (briandealwis) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/META-INF/MANIFEST.MF]: This name doesn't match the directory and pom.xml artifact name: it's com.gcp.**eclipse**.appengine.localserver.test

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+Bundle-Vendor: Google, Inc.
+Bundle-Version: 94.0.0.qualifier
+Fragment-Host: com.google.gcp.appengine.localserver
```
Comment (briandealwis) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/META-INF/MANIFEST.MF]: The actual host bundle is com.gcp.**eclipse**.appengine.localserver.test

**Comment by briandealwis** [comment]
@elharo: you're missing the **eclipse** in the fragment and fragment-host: it's com.google.gcp.**eclipse**.appengine.localserver

**Comment by elharo** [comment]
You know, I looked for that, repeatedly, and I still couldn't see the bug. :-) Thanks.

**Comment by elharo** [comment]
Getting closer. Now the problem is 

[INFO] Expected eclipse log file: /Users/elharo/gcloud-eclipse-tools/plugins/com.google.gcp.eclipse.appengine.localserver.test/target/work/data/.metadata/.log
[INFO] Command line:
	[/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre/bin/java, -Dosgi.noShutdown=false, -Dosgi.os=macosx, -Dosgi.ws=cocoa, -Dosgi.arch=x86_64, -Dorg.eclipse.swtbot.search.timeout=10000, -Xms40m, -Xmx1G, ${os-jvm-flags}, -Dosgi.clean=true, -jar, /Users/elharo/.m2/repository/p2/osgi/bundle/org.eclipse.equinox.launcher/1.3.100.v20150511-1540/org.eclipse.equinox.launcher-1.3.100.v20150511-1540.jar, -data, /Users/elharo/gcloud-eclipse-tools/plugins/com.google.gcp.eclipse.appengine.localserver.test/target/work/data, -install, /Users/elharo/gcloud-eclipse-tools/plugins/com.google.gcp.eclipse.appengine.localserver.test/target/work, -configuration, /Users/elharo/gcloud-eclipse-tools/plugins/com.google.gcp.eclipse.appengine.localserver.test/target/work/configuration, -application, org.eclipse.tycho.surefire.osgibooter.headlesstest, -testproperties, /Users/elharo/gcloud-eclipse-tools/plugins/com.google.gcp.eclipse.appengine.localserver.test/target/surefire.properties, -testApplication, org.eclipse.ui.ide.workbench, -product, org.eclipse.sdk.ide]
Error: Could not find or load main class ${os-jvm-flags}

**Comment by elharo** [comment]
@briandealwis Could you please take another look at this? There seems to be some aspect of configuration for eclipse test bundles with tycho I'm missing. Thanks.

**Comment by elharo** [comment]
NVM, got it working. Real test co,ing soon.

**Comment by elharo** [comment]
@akerekes @briandealwis Please review.

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/src/com/google/gcp/eclipse/appengine/localserver/XmlTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+public class XmlTest {
+    
+  private DocumentBuilder builder;
+	
+@Before
+  public void setUp() throws ParserConfigurationException {
+    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+    factory.setNamespaceAware(true);
+	builder = factory.newDocumentBuilder();
```
Comment (akerekes) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/src/com/google/gcp/eclipse/appengine/localserver/XmlTest.java]: minor: indentation

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/build.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+source.. = src/
+output.. = bin/
```
Comment (akerekes) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/build.properties]: should this go to bin or target?

**Comment by briandealwis** [comment]
Verified the tests are run and pass with Tycho.

**Comment by briandealwis** [comment]
We're committing .project files, right?  Missing `plugins/com.google.gcp.eclipse.appengine.localserver.test/.project`

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/build.properties (hunk 4 above)
Comment (briandealwis) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/build.properties]: Good catch: it has to match the output location in `.classpath`

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/src/com/google/gcp/eclipse/appengine/localserver/XmlTest.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.w3c.dom.Document;
+
+public class XmlTest {
```
Comment (briandealwis) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/src/com/google/gcp/eclipse/appengine/localserver/XmlTest.java]: I've never seen a unit test to validate XML files. Nice idea :+1:

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/src/com/google/gcp/eclipse/appengine/localserver/XmlTest.java (hunk 3 above)
Comment (elharo) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/src/com/google/gcp/eclipse/appengine/localserver/XmlTest.java]: done

**Comment by elharo** [comment]
.project included now

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/build.properties (hunk 4 above)
Comment (elharo) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/build.properties]: done

**Comment by briandealwis** [comment]
We need to disable `useJDK` on Travis, which is ok since we're ensuring we build with `jdk7`

File: plugins/com.google.gcp.eclipse.appengine.localserver.test/build.properties [hunk 6]
```diff
@@ -1,5 +1,5 @@
 source.. = src/
-output.. = bin/
+output.. = target/
```
Comment (briandealwis) [inline plugins/com.google.gcp.eclipse.appengine.localserver.test/build.properties]: This should be `target/classes`, and it's needed in `plugins/com.google.gcp.eclipse.appengine.localserver` too.

**Comment by briandealwis** [comment]
lgtm

**Comment by briandealwis** [comment]
Travis failed on its `install` step; it doesn't actually help us, so we'll skip it.

**Pull request merged**